src/astro/databases/google/bigquery.py

```diff
@@ -236,3 +257,177 @@ def gs_to_bigquery(
         self.hook.insert_job(
             configuration=job_config,
         )
+
+    def s3_to_bigquery(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        Transfer data from S3 to Bigquery via invoking datatransfer job
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+        """
+        native_support_kwargs = native_support_kwargs or {}
+
+        project_id = self.get_project_id(target_table)
+
+        if if_exists == "replace":
+            # We need to create an empty table as datatransfer job can only append to an existing table,
+            # so we need to create a table.
+            self.create_empty_table(source_file, target_table)
+
+        transfer = S3ToBigqueryDataTransfer(
+            target_table=target_table,
+            source_file=source_file,
+            project_id=project_id,
+            native_support_kwargs=native_support_kwargs,
+            **kwargs,
+        )
+        transfer.run()
+
+
+class S3ToBigqueryDataTransfer:
+    """
+    Create and run Datatransfer job from S3 to Bigquery
+    """
+
+    def __init__(
+        self,
+        target_table: Table,
+        source_file: File,
+        project_id: str,
+        poll_duration: int = 1,
+        native_support_kwargs: Optional[Dict] = None,
+        **kwargs,
+    ):
+        """
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param project_id: Bigquery project id
+        :param poll_duration: sleep duration between two consecutive job status checks. Unit - seconds. Default 1 sec.
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
+        """
+        self.client = BiqQueryDataTransferServiceHook(gcp_conn_id=target_table.conn_id)
+        self.target_table = target_table
+        self.source_file = source_file
+
+        conn = BaseHook.get_connection(source_file.conn_id)
+        self.s3_login = conn.login
+        self.s3_password = conn.password
+        self.s3_file_type = NATIVE_PATHS_SUPPORTED_FILE_TYPES.get(source_file.type.name)
+
+        self.project_id = project_id
+        self.poll_duration = poll_duration
+        self.native_support_kwargs = native_support_kwargs
+        self.kwargs = kwargs
+
+    def run(self):
+        """
+        Algo to run S3 to Bigquery datatransfer
+        """
+        transfer_config_id = self.create_transfer_config()
+        try:
```

Comment:
Why don't we have an `except` clause? Are we at the risk of silencing possible issues?

Reply:
We are not silencing possible issues, they will be raised for sure. But the intent of the code is to make sure we delete the transfer config once it's created, regardless of any issues.

![Screenshot 2022-07-13 at 12 38 39 AM](https://user-images.githubusercontent.com/13021213/178574692-fe72b712-c079-435e-b6b5-d17029cdbae4.png)